### PR TITLE
WT-5583 Fix a dead lock in eviction server flow

### PR DESCRIPTION
**Pull request by kommiharibabu**
In the eviction server code flow when there are no eviction workers,
the server itself try to evict the pages, during the page eviction
flow it try to take the lock which itself is already held leads to
a dead lock.

Fix is to release the lock before evicting the page from the queue
and take back the lock once it finish processing the queue.

### CODE REVIEW

File: src/evict/evict_lru.c [hunk 1]
```diff
@@ -700,9 +701,21 @@ __evict_pass(WT_SESSION_IMPL *session)
         if (!WT_EVICT_HAS_WORKERS(session) &&
           (cache->evict_empty_score < WT_EVICT_SCORE_CUTOFF ||
             !__evict_queue_empty(cache->evict_urgent_queue, false))) {
+            /*
+             * Release the evict pass lock as the thread is going to evict the pages from the queue.
+             * Otherwise, it can lead to a dead lock while evicting the page the lead to clear the
```
Comment (quchenhao) [inline src/evict/evict_lru.c]: that lead to clear the?

File: test/csuite/Makefile.am [hunk 2]
```diff
@@ -27,17 +26,15 @@ all_TESTS += test_rwlock
...
 
 test_scope_SOURCES = scope/main.c
 noinst_PROGRAMS += test_scope
 all_TESTS += test_scope
 
 test_timestamp_abort_SOURCES = timestamp_abort/main.c
 noinst_PROGRAMS += test_timestamp_abort
-# Temporarily disabled
-# all_TESTS += timestamp_abort/smoke.sh
+-all_TESTS += timestamp_abort/smoke.sh
```
Comment (quchenhao) [inline test/csuite/Makefile.am]: -all_TESTS?

File: src/evict/evict_lru.c (hunk 1 above)
Comment (kommiharibabu) [reply src/evict/evict_lru.c]: Fixed.

File: test/csuite/Makefile.am (hunk 2 above)
Comment (kommiharibabu) [reply test/csuite/Makefile.am]: Fixed.

**Comment by kommiharibabu** [comment]
The failed test is not related to the changes in the patch. There is already a ticket raised to the failed test.

**Pull request merged**